Sources/GRPCCore/Call/Server/ServerInterceptorOperation.swift

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// A `ServerInterceptorOperation` describes to which RPCs a server interceptor should be applied.
+///
+/// You can configure a server interceptor to be applied to:
+/// - all RPCs and services;
+/// - requests directed only to specific services registered with your server; or
+/// - requests directed only to specific methods (of a specific service).
+///
+/// - SeeAlso: ``ServerInterceptor`` for more information on server interceptors, and
+///  ``ClientInterceptorOperation`` for the client-side version of this type.
+public struct ServerInterceptorOperation: Sendable {
+  internal enum Wrapped: Sendable {
+    case allServices(interceptor: any ServerInterceptor)
+    case serviceSpecific(interceptor: any ServerInterceptor, services: [String])
+    case methodSpecific(interceptor: any ServerInterceptor, methods: [MethodDescriptor])
+  }
```

Comment:
The distinguishing part is what the interceptor applies to, rather than the combination so I'd probably represent this as:

```swift
public struct ServerInterceptorOperation {
  enum AppliesTo {
    case all
    case services([ServiceDescriptor])
    case methods([MethodDescriptor])
  }

  var interceptor: any ServerInterceptor
  var appliesTo: AppliesTo
}
```

This also lets you remove the computer interceptor property on your `Wrapped` type

Reply:
Thinking about this more, it might actually make sense to promote the `AppliesTo` to the API, then you can reduce the number of static funcs on `ServerInterceptorOperation` to one:

```swift
public struct ServerInterceptorOperation {
  public static func apply(_ interceptor: any ServerInteceptor, to subject: AppliesTo) -> Self
}
```

So then we calling it reads quite naturally:

```swift
interceptorPipeline: [
  .apply(foo, to: .all),
  .apply(bar, to: .services([...])
  .apply(baz, to: .methods([...])
]
```

Reply:
Nice one! I agree it reads more nicely